libxrdp/libxrdpinc.h

```diff
@@ -22,6 +22,8 @@
 #define LIBXRDPINC_H
 
 #include "xrdp_rail.h"
+#include <stdint.h>
+#include "list.h"
```

Comment:
Like you, I absolutely believe that includes should stand on their own as far as declaring what they need to compile, and shouldn't depend on previous includes.

`<stdint.h>` doesn't appear to be required here and can be removed. Also, the only references to the list.h include are a  `struct list *`, and consequently you don't need to include the whole file. An incomplete type declaration will do here (i.e. `struct list;`)

Reply:
Good points. <stdint> was a leftover from a previously interrupted debugging session.

I think it's reasonable to include just the `struct list;` directive.